src/main/kotlin/tech/thatgravyboat/skyblockpv/screens/tabs/farming/ComposterScreen.kt

```diff
@@ -0,0 +1,270 @@
+                map.forEach {
+                    it.fill(value)
+                }
+            }
+
+            loading(
+                gardenProfile,
+                onSuccess = { data ->
+                    val staticPlots = StaticGardenData.plots.toMutableList().apply { removeAll(data.unlockedPlots) }
+                    data.unlockedPlots.forEach {
+                        map[it.location] = Displays.tooltip(
+                            Displays.item(Items.GREEN_STAINED_GLASS_PANE),
+                            it.getName().also { it.color = TextColor.GREEN },
+                        )
+                    }
+                    val unlockedAmount = data.unlockedPlots.groupBy { it.type }.mapValues { it.value.size }
+                    staticPlots.forEach {
+                        val plots = unlockedAmount[it.type] ?: 0
+                        val cost = StaticGardenData.plotCost[it.type] ?: emptyList()
+
+                        if (plots >= cost.size) {
+                            return@forEach
+                        }
+
+                        val plotCost = cost[plots]
+                        map[it.location] = Displays.item(Items.BLACK_STAINED_GLASS_PANE).withTooltip(
+                            it.getName(),
+                            plotCost.getDisplay().copy().apply { append(Text.of(" x${plotCost.amount}") { color = TextColor.DARK_GRAY }) },
+                        )
+                    }
+                },
+                loadingValue = {
+                    fillMap(
+                        Displays.tooltip(
+                            Displays.item(Items.ORANGE_STAINED_GLASS_PANE),
+                            Text.of("Loading...") { this.color = TextColor.GOLD },
+                        ),
+                    )
+                },
+                errorValue = {
+                    fillMap(
+                        Displays.tooltip(
+                            Displays.item(Items.BEDROCK),
+                            Text.of("Error!") { this.color = TextColor.RED },
+                        ),
+                    )
+                },
+            )
+
+            map[2][2] = Displays.tooltip(
+                Displays.item(
+                    loaded(
+                        gardenProfile,
+                        Items.BARRIER.defaultInstance,
+                        Items.BEDROCK.defaultInstance,
+                    ) { it.selectedBarnSkin.getItem() },
+                ),
+                loadingComponent(gardenProfile) { it.selectedBarnSkin.displayName },
+            )
+
+            val plots = map.map { it.reversed().map { Displays.padding(2, it) }.toColumn() }.toRow()
+            display(
+                Displays.background(
+                    SkyBlockPv.id("inventory/inventory-5x5"),
+                    Displays.padding(2, plots),
+                ),
+            )
+        },
+    )
+
+    private operator fun <E> MutableList<MutableList<E>>.set(location: Vector2i, value: E) {
```

Comment:
this is seemingly not used

Reply:
![image](https://github.com/user-attachments/assets/f8688878-d0fb-481b-a539-73e89450e2f6)